polygeist/test/polygeist-opt/sycl/loop_internalization.mlir

```diff
@@ -315,132 +315,133 @@ gpu.func @kernel(%arg0: memref<?x!sycl_accessor_3_f32_r_gb>, %arg1: memref<?x!sy
-// CHECK-NEXT:            %[[VAL_61:.*]] = memref.load %[[VAL_60]]{{\[}}%[[VAL_59]]] : memref<?xf32, 1>
-// CHECK-NEXT:            memref.store %[[VAL_61]], %[[VAL_49]]{{\[}}%[[LOCALID0]], %[[LOCALID1]]] : memref<?x?xf32, #sycl.access.address_space<local>>
+// CHECK-NEXT:            %[[VAL_54:.*]] = memref.alloca() : memref<1x!sycl_id_2_>
+// CHECK-NEXT:            %[[VAL_55:.*]] = arith.constant 0 : index
+// CHECK-NEXT:            %[[VAL_56:.*]] = arith.constant 0 : i32
+// CHECK-NEXT:            %[[VAL_57:.*]] = sycl.id.get %[[VAL_54]]{{\[}}%[[VAL_56]]] : (memref<1x!sycl_id_2_>, i32) -> memref<?xindex>
+// CHECK-NEXT:            memref.store %[[VAL_35]], %[[VAL_57]]{{\[}}%[[VAL_55]]] : memref<?xindex>
+// CHECK-NEXT:            %[[VAL_58:.*]] = arith.constant 1 : i32
+// CHECK-NEXT:            %[[VAL_59:.*]] = sycl.id.get %[[VAL_54]]{{\[}}%[[VAL_58]]] : (memref<1x!sycl_id_2_>, i32) -> memref<?xindex>
+// CHECK-NEXT:            memref.store %[[VAL_47]], %[[VAL_59]]{{\[}}%[[VAL_55]]] : memref<?xindex>
+// CHECK-NEXT:            %[[VAL_60:.*]] = arith.constant 0 : index
+// CHECK-NEXT:            %[[VAL_61:.*]] = sycl.accessor.subscript %[[VAL_0]]{{\[}}%[[VAL_54]]] : (memref<?x!sycl_accessor_2_f32_r_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
+// CHECK-NEXT:            %[[VAL_62:.*]] = memref.load %[[VAL_61]]{{\[}}%[[VAL_60]]] : memref<?xf32, 1>
+// CHECK-NEXT:            memref.store %[[VAL_62]], %[[VAL_50]]{{\[}}%[[LOCALID0]], %[[LOCALID1]]] : memref<?x?xf32, #sycl.access.address_space<local>>
 
 // COM: Calculate offset:
-// CHECK-NEXT:            %[[VAL_62:.*]] = arith.constant 4 : index
-// CHECK-NEXT:            %[[VAL_63:.*]] = arith.muli %[[VAL_62]], %[[WGSIZE0]] : index
-// CHECK-NEXT:            %[[VAL_64:.*]] = arith.muli %[[VAL_63]], %[[WGSIZE1]] : index
-// CHECK-NEXT:            %[[VAL_65:.*]] = arith.addi %[[VAL_48]], %[[VAL_64]] : index
+// CHECK-NEXT:            %[[VAL_63:.*]] = arith.constant 4 : index
+// CHECK-NEXT:            %[[VAL_64:.*]] = arith.muli %[[VAL_63]], %[[WGSIZE0]] : index
+// CHECK-NEXT:            %[[VAL_65:.*]] = arith.muli %[[VAL_64]], %[[WGSIZE1]] : index
+// CHECK-NEXT:            %[[VAL_66:.*]] = arith.addi %[[VAL_49]], %[[VAL_65]] : index
 
-// COM: Get pointer to the  shared local memory portion for 2nd memref:
-// CHECK-NEXT:            %[[VAL_66:.*]] = memref.view %[[VAL_41]]{{\[}}%[[VAL_65]]]{{\[}}%[[WGSIZE0]], %[[WGSIZE1]]] : memref<32000xi8, #sycl.access.address_space<local>> to memref<?x?xf32, #sycl.access.address_space<local>>
+// COM: Get pointer to the shared local memory portion for 2nd memref:
+// CHECK-NEXT:            %[[VAL_67:.*]] = memref.view %[[VAL_42]]{{\[}}%[[VAL_66]]]{{\[}}%[[WGSIZE1]], %[[WGSIZE0]]] : memref<32000xi8, #sycl.access.address_space<local>> to memref<?x?xf32, #sycl.access.address_space<local>>
```

Comment:
Since the indexes is now changed to `[LOCALID1][LOCALID0]`, sizes should also be changed to `[WGSIZE1], [WGSIZE0]`.

Reply:
You mean that `WGSIZE0` and `WGSIZE1` now are swapped ?

Reply:
yes, they are swapped, because the indexes of `view` is swapped.